dao/src/main/java/org/thingsboard/server/dao/alarm/BaseAlarmService.java

```diff
@@ -312,20 +313,28 @@ private ListenableFuture<PageData<AlarmInfo>> fetchAlarmsOriginators(TenantId te
         List<ListenableFuture<AlarmInfo>> alarmFutures = new ArrayList<>(alarms.getData().size());
         for (AlarmInfo alarmInfo : alarms.getData()) {
             alarmFutures.add(Futures.transform(
-                    entityService.fetchEntityNameAsync(tenantId, alarmInfo.getOriginator()), originatorName -> {
-                        if (originatorName == null) {
-                            originatorName = "Deleted";
-                        }
-                        alarmInfo.setOriginatorName(originatorName);
-                        return alarmInfo;
-                    }, MoreExecutors.directExecutor()
+                    entityService.fetchEntityHasNameAsync(tenantId, alarmInfo.getOriginator()), originatorHasName ->
+                            updateAlarmInfo(alarmInfo, originatorHasName),
+                    MoreExecutors.directExecutor()
             ));
         }
         return Futures.transform(Futures.successfulAsList(alarmFutures),
                 alarmInfos -> new PageData<>(alarmInfos, alarms.getTotalPages(), alarms.getTotalElements(),
                         alarms.hasNext()), MoreExecutors.directExecutor());
     }
 
+    private AlarmInfo updateAlarmInfo(AlarmInfo alarmInfo, HasName hasName) {
```

Comment:
For backward compatibility, let's set the originatorName to "Deleted" when fetchAlarmsOriginators method is called. In other cases, let's set "N/A".

Reply:
@ashvayka, I'm not sure we should return "Deleted" when the name of alarm originator is null. There might be use cases where the alarm was created for entity that doesn't have a name. Should we update this to set "N/A" instead of "Deleted"? Also, we have a case where we do not even check for name value, which means null will be set if there is no name. See the example in the findAlarmInfoByIdAsync method. Should these methods be updated as well? Or should we leave them unchanged for backward compatibility?